File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commercechannel/CommercechannelService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commercechannel;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommercechannelService extends BaseService {

	public CommercechannelService(Session session) {
		super(session);
	}

	public JSONObject addCommerceChannel(long siteGroupId, String name, String type, JSONObject typeSettingsUnicodeProperties, String commerceCurrencyCode, String externalReferenceCode, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCurrencyCode", checkNull(commerceCurrencyCode));
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));
			_params.put("name", checkNull(name));
			_params.put("siteGroupId", siteGroupId);
			_params.put("type", checkNull(type));
			_params.put("typeSettingsUnicodeProperties", checkNull(typeSettingsUnicodeProperties));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commercechannel/add-commerce-channel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject deleteCommerceChannel(long commerceChannelId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceChannelId", commerceChannelId);

			_command.put("/commerce.commercechannel/delete-commerce-channel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject fetchByExternalReferenceCode(long companyId, String externalReferenceCode) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));

			_command.put("/commerce.commercechannel/fetch-by-external-reference-code", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject fetchCommerceChannel(long commerceChannelId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceChannelId", commerceChannelId);

			_command.put("/commerce.commercechannel/fetch-commerce-channel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getCommerceChannel(long commerceChannelId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceChannelId", commerceChannelId);

			_command.put("/commerce.commercechannel/get-commerce-channel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getCommerceChannelByOrderGroupId(long groupId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);

			_command.put("/commerce.commercechannel/get-commerce-channel-by-order-group-id", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceChannels(int start, int end) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("end", end);
			_params.put("start", start);

			_command.put("/commerce.commercechannel/get-commerce-channels", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray getCommerceChannels(long companyId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);

			_command.put("/commerce.commercechannel/get-commerce-channels", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray searchCommerceChannels(long companyId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);

			_command.put("/commerce.commercechannel/search-commerce-channels", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public JSONArray searchCommerceChannels(long companyId, String keywords, int start, int end, JSONObjectWrapper sort) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("end", end);
			_params.put("keywords", checkNull(keywords));
			_params.put("start", start);
			mangleWrapper(_params, "sort", "com.liferay.portal.kernel.search.Sort", sort);

			_command.put("/commerce.commercechannel/search-commerce-channels", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer searchCommerceChannelsCount(long companyId, String keywords) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("companyId", companyId);
			_params.put("keywords", checkNull(keywords));

			_command.put("/commerce.commercechannel/search-commerce-channels-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject updateCommerceChannel(long commerceChannelId, long siteGroupId, String name, String type, JSONObject typeSettingsUnicodeProperties, String commerceCurrencyCode) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceChannelId", commerceChannelId);
			_params.put("commerceCurrencyCode", checkNull(commerceCurrencyCode));
			_params.put("name", checkNull(name));
			_params.put("siteGroupId", siteGroupId);
			_params.put("type", checkNull(type));
			_params.put("typeSettingsUnicodeProperties", checkNull(typeSettingsUnicodeProperties));

			_command.put("/commerce.commercechannel/update-commerce-channel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceChannel(long commerceChannelId, long siteGroupId, String name, String type, JSONObject typeSettingsUnicodeProperties, String commerceCurrencyCode, String priceDisplayType, boolean discountsTargetNetPrice) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceChannelId", commerceChannelId);
			_params.put("commerceCurrencyCode", checkNull(commerceCurrencyCode));
			_params.put("discountsTargetNetPrice", discountsTargetNetPrice);
			_params.put("name", checkNull(name));
			_params.put("priceDisplayType", checkNull(priceDisplayType));
			_params.put("siteGroupId", siteGroupId);
			_params.put("type", checkNull(type));
			_params.put("typeSettingsUnicodeProperties", checkNull(typeSettingsUnicodeProperties));

			_command.put("/commerce.commercechannel/update-commerce-channel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCommerceChannelExternalReferenceCode(long commerceChannelId, String externalReferenceCode) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceChannelId", commerceChannelId);
			_params.put("externalReferenceCode", checkNull(externalReferenceCode));

			_command.put("/commerce.commercechannel/update-commerce-channel-external-reference-code", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Commercechannel/CommercechannelService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commercechannel
{
	public class CommercechannelService : ServiceBase
	{
		public CommercechannelService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> GetCommerceChannelAsync(long commerceChannelId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceChannelId", commerceChannelId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/get-commerce-channel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceChannelAsync(long siteGroupId, string name, string type, IDictionary<string, object> typeSettingsUnicodeProperties, string commerceCurrencyCode, string externalReferenceCode, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("siteGroupId", siteGroupId);
			_parameters.Add("name", name);
			_parameters.Add("type", type);
			_parameters.Add("typeSettingsUnicodeProperties", typeSettingsUnicodeProperties);
			_parameters.Add("commerceCurrencyCode", commerceCurrencyCode);
			_parameters.Add("externalReferenceCode", externalReferenceCode);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/add-commerce-channel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> DeleteCommerceChannelAsync(long commerceChannelId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceChannelId", commerceChannelId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/delete-commerce-channel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchByExternalReferenceCodeAsync(long companyId, string externalReferenceCode)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("externalReferenceCode", externalReferenceCode);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/fetch-by-external-reference-code", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> FetchCommerceChannelAsync(long commerceChannelId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceChannelId", commerceChannelId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/fetch-commerce-channel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCommerceChannelByOrderGroupIdAsync(long groupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/get-commerce-channel-by-order-group-id", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceChannelsAsync(int start, int end)
		{
			var _parameters = new JsonObject();

			_parameters.Add("start", start);
			_parameters.Add("end", end);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/get-commerce-channels", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceChannelsAsync(long companyId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/get-commerce-channels", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> SearchCommerceChannelsAsync(long companyId, string keywords, int start, int end, JsonObjectWrapper sort)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("keywords", keywords);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "sort", "$languageUtil.getJSONWrapperClassName($parameter.type)", sort);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/search-commerce-channels", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<IEnumerable<dynamic>> SearchCommerceChannelsAsync(long companyId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/search-commerce-channels", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> SearchCommerceChannelsCountAsync(long companyId, string keywords)
		{
			var _parameters = new JsonObject();

			_parameters.Add("companyId", companyId);
			_parameters.Add("keywords", keywords);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/search-commerce-channels-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCommerceChannelAsync(long commerceChannelId, long siteGroupId, string name, string type, IDictionary<string, object> typeSettingsUnicodeProperties, string commerceCurrencyCode, string priceDisplayType, bool discountsTargetNetPrice)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceChannelId", commerceChannelId);
			_parameters.Add("siteGroupId", siteGroupId);
			_parameters.Add("name", name);
			_parameters.Add("type", type);
			_parameters.Add("typeSettingsUnicodeProperties", typeSettingsUnicodeProperties);
			_parameters.Add("commerceCurrencyCode", commerceCurrencyCode);
			_parameters.Add("priceDisplayType", priceDisplayType);
			_parameters.Add("discountsTargetNetPrice", discountsTargetNetPrice);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/update-commerce-channel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCommerceChannelAsync(long commerceChannelId, long siteGroupId, string name, string type, IDictionary<string, object> typeSettingsUnicodeProperties, string commerceCurrencyCode)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceChannelId", commerceChannelId);
			_parameters.Add("siteGroupId", siteGroupId);
			_parameters.Add("name", name);
			_parameters.Add("type", type);
			_parameters.Add("typeSettingsUnicodeProperties", typeSettingsUnicodeProperties);
			_parameters.Add("commerceCurrencyCode", commerceCurrencyCode);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/update-commerce-channel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdateCommerceChannelExternalReferenceCodeAsync(long commerceChannelId, string externalReferenceCode)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceChannelId", commerceChannelId);
			_parameters.Add("externalReferenceCode", externalReferenceCode);

			var _command = new JsonObject()
			{
				{ "/commerce.commercechannel/update-commerce-channel-external-reference-code", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}